src/NetTopologySuite.IO.Oracle/OracleGeometryWriter.cs

```diff
@@ -40,6 +41,8 @@ public SdoGeometry Write(Geometry geometry)
                 return null;
             }
 
+            this.dimension = Dimension(geometry);
```

Comment:
I don't like this as it messes things up in a re-entrant scenario. Make `dimension` a local variable and pass it to the private specific `Write` functions.